lib/cli.rb

```diff
@@ -233,9 +237,9 @@ def define_options(parser)
     end
 
     def parse(args)
-      @parser.order!(args)
-      @mode = args.shift
-      @sub_parser[@mode]&.order!(args) unless @mode.nil?
+      left = @parser.order(args)
+      @mode = left.shift
+      @sub_parser[@mode]&.order!(left) unless @mode.nil?
```

Comment:
why do we need this? behaviour is the same

Reply:
`@parser.order!` modifies `args`, which is problematic because we use `args` again when entering a namespace.